metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -56,20 +56,23 @@ void build_successor(const std::string &graph_fname,
 
     // terminal uses BOSS edges as indices, so we need to map it to annotation indices
     sdsl::bit_vector term(graph.num_nodes(), 0);
+    uint64_t num_anchors = 0;
     for (uint64_t i = 1; i < terminal.size(); ++i) {
         if (terminal[i]) {
             uint64_t graph_idx = graph.boss_to_kmer_index(i);
             uint64_t anno_index = graph::AnnotatedSequenceGraph::graph_to_anno_index(
                     graph_idx);
             assert(anno_index < graph.num_nodes());
             term[anno_index] = 1;
+            num_anchors++;
         }
     }
+    logger->trace("Number of anchors before anchor optimization: {}", num_anchors);
```

Comment:
maybe just `sdsl::util::cnt_one_bits(term)` -- it's extremely fast, you'll save two lines of code